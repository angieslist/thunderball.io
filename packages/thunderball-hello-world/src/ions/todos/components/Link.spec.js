import React from 'react';
import { shallow } from 'enzyme';

import Link from './Link';

const setup = (setupProps = {}) => {
  const defaultProps = {
    active: false,
    onClick: jest.fn(),
    children: 'Test link',
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <Link {...props}>
      {props.children}
    </Link>,
  );

  return {
    props,
    wrapper,
  };
};

describe('Link', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('renders a span when active is true', () => {
    const { wrapper } = setup({ active: true });
    expect(wrapper).toMatchSnapshot();
  });

  test('calls onClick() on click', () => {
    const { props, wrapper } = setup();

    const link = wrapper.find('button');
    link.simulate('click');

    expect(props.onClick).toHaveBeenCalled();
  });
});
