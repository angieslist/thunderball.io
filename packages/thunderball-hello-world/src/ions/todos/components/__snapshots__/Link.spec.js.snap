// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders a span when active is true 1`] = `
<button
  disabled={true}
  onClick={[MockFunction]}
  style={
    Object {
      "marginLeft": "4px",
    }
  }
>
  Test link
</button>
`;

exports[`Link renders without crashing 1`] = `
<button
  disabled={false}
  onClick={[MockFunction]}
  style={
    Object {
      "marginLeft": "4px",
    }
  }
>
  Test link
</button>
`;
